src/moscot/problems/time/_mixins.py

```diff
@@ -768,4 +769,9 @@ def temporal_key(self) -> Optional[str]:
     def temporal_key(self: TemporalMixinProtocol[K, B], key: Optional[str]) -> None:
         if key is not None and key not in self.adata.obs:
             raise KeyError(f"Unable to find temporal key in `adata.obs[{key!r}]`.")
+        if key is not None and not is_numeric_dtype(self.adata.obs[key]):
+            raise TypeError(
+                "`temporal_key` has to be of numeric type."
+                + f"Found `adata.obs[{key!r}]` to be of type {infer_dtype(self.adata.obs[key])}."
```

Comment:
No need to use `+`, also missing `' '` after `...numeric type.`
Pet peeve: woul refer to the temporal key as `Temporal key`, not \`temporal_key\`.